src/Storages/StorageS3.cpp

```diff
@@ -460,7 +460,7 @@ class StorageS3Source::KeysIterator::Impl : WithContext
         for (auto && key : all_keys)
         {
             std::optional<S3::ObjectInfo> info;
-            if (need_total_size)
+            if (need_total_size && all_keys.size() == 1)
```

Comment:
Why can't we pass `need_total_size = false` in that case?

Reply:
because it is passed from `StorageS3::read` when we do `createFileIterator`, but StorageIceberg/DeltaLake/Hudi do not implement the `read` method at all (as they inherit from `StorageS3`, `StorageS3::read` is directly called).